enhancements/network/ovn-multi-external-gateway.md

```diff
@@ -0,0 +1,182 @@
+- Extend the existing unit tests for the annotation based logic to accommodate for the scenario where CR and annotations exist in 4.13 and onwards. 
+
+### Non-Goals
+
+- Functional improvements on the logic to handle multiple external gateways that are not supported in the existing logic.
+- Removal of the annotation based logic to configure multiple external gateways. This should be carrier out at least no sooner than OpenShift 4.15 to guarantee a stable deprecation path to existing deployments that leverage on the annotations to configure the external gateways.
+
+## Proposal
+
+Create a new CRD that contains a direct relationship to the annotations used in the pod/namespace to manage the external gateways configuration:
+
+Namespace annotations:
+* `k8s.ovn.org/routing-external-gws`
+* `k8s.ovn.org/external-gw-pod-ips`
+
+Pod annotations:
+* `k8s.ovn.org/routing-namespaces`
+* `k8s.ovn.org/routing-network`
+* `k8s.ovn.org/bfd-enabled`
+
+The implementation will also include a new controller that will watch cluster wide for instances of this CRD and apply the changes to the OVN network configuration as it is currently done by the annotation based logic.
+
+The controller will use the OpenShift release version as well as the existence of the annotations in the namespace and pods to apply the changes to the network configuration:
+
+| **OpenShift version** | **Annotations exist** | **CR exists** | **Action**                                        |
+|-----------------------|-----------------------|---------------|---------------------------------------------------|
+| 4.13                  | No                    | Yes           | Use CR information to configure external gateways |
+| 4.13                  | Yes                   | Yes           | Use annotations to configure external gateways    |
+| 4.13                  | Yes                   | No            | Use annotations to configure external gateway     |
+| 4.13                  | No                    | No            | Nothing to do                                     |
+| 4.14                  | No                    | Yes           | Use CR information to configure external gateways |
+| 4.14                  | Yes                   | Yes           | Use CR information to configure external gateways |
+| 4.14                  | Yes                   | No            | Use annotations to configure external gateways    |
+| 4.14                  | No                    | No            | Nothing to do                                     |
+
+The current annotation based logic will be enhanced as part of this change to support the logic described in the previous table.
+
+### Workflow Description
+The workflow follows the behavior of a namespaced CR.
+
+
+### API Extensions
+
+A new namespaced-scoped CRD is created to represent the multiple external gateways configuration. The `spec` structure exposes the following fields:
```

Comment:
Why is this a namespace-scoped CRD? I'd assume this whole feature is not for users/app owners, more for admins.

Reply:
on second thoughts to make the below CRD neater, we can make this namespace-scoped but then the scoped namespace should be the one on which the config is applied to, that might mean a lot of crds, one per namespace.